src/transaction/log_tran_table.c

```diff
@@ -4157,6 +4157,8 @@ logtb_complete_mvcc (THREAD_ENTRY * thread_p, LOG_TDES * tdes, bool committed)
 
   logtb_tran_clear_update_stats (&tdes->log_upd_stats);
 
+  tdes->last_mvcc_lsa.set_null ();
```

Comment:
Move the line in the `if (MVCCID_IS_VALID (mvccid))` branch and assert that it is null on the else - mvcc not valid - branch.
Also, shouldn't this change be added to all the places where `mvccinfo.reset ()` is called, in:
- `logtb_initialize_system_tdes`
- `logtb_release_tran_index`

Reply:
I agree with your first suggestion. I will move it and add assert. 

> logtb_initialize_system_tdes
logtb_release_tran_index

logtb_clear_tdes (), which initialize the last_mvcc_lsa, is called in those functions.

Reply:
Agree. Indeed, it is called in the first one.

Reply:
For logtb_release_tran_index (), 
```
TR_TABLE_CS_ENTER () 
logtb_release_tran_index ()
-> logtb_free_tran_index ()
-> logtb_clear_tdes ()
TR_TABLE_CS_EXIT ()
```
For logtb_initialize_system_tdes ()
```
logtb_initialize_system_tdes ()
{
	logtb_clear_tdes ()
	tdes->mvccinfo.reset ()
}
```

I don't think it needs to be added to above places where mvccinfo.reset () is called.

However, last_mvcc_lsa seems to be used whenever dealing with mvcc information, so I wonder if it would be better to move last_mvcc_lsa to tdes->mvccinfo and initialize it together in the mvccinfo.reset() function. Not now, it is just thought.

Reply:
I agree with @joohok that it should be packed together if it's used and initialized together all the time.

Reply:
I also thought of this, but did not voice my concern.
Then let's do it now.
Either with this patch, or a new patch on the same issue.
Or, if you prefer, add a new Jira issue.
If this came up as a possible improvement, I don't want to lose track of it.

Reply:
OK, I will do it with next patch on the same issue.